TypeScript error when trying to provide a stateHandler with setup and teardown

### Software versions

_Please provide at least OS and version of pact-js_

- **OS**: Arch Linux, Ubuntu, Windows (it's irrelevant)
- **Consumer Pact library**: Pact JS v12.1.2
- **Provider Pact library**: Pact JS v12.1.2
- **Node Version**: v20.11.0

### Issue Checklist

Please confirm the following:

- [x] I have upgraded to the latest
- [x] I have the read the FAQs in the Readme
- [x] I have triple checked, that there are **no unhandled promises** in my code and have [read](https://github.com/pact-foundation/pact-js/blob/master/docs/troubleshooting.md#test-fails-when-it-should-pass) the section on intermittent test failures
- [x] I have set my log level to debug and attached a log file showing the complete request/response cycle
- [ ] For bonus points and virtual high fives, I have created a reproduceable git repository (see below) to illustrate the problem

### Expected behaviour

There are no TypeScript errors when trying to provide a stateHandler with setup and teardown to a `Verifier`.

### Actual behaviour

Trying to provide a stateHandler with setup and teardown results in the following TypeScript error:
```
error TS2322: Type '{ setup: () => Promise<void>; teardown: () => Promise<void>; }' is not assignable to type 'StateHandler & ((state: string, params?: { [name: string]: string; } | undefined) => Promise<unknown>)'.
  Type '{ setup: () => Promise<void>; teardown: () => Promise<void>; }' is not assignable to type 'StateFuncWithSetup & ((state: string, params?: { [name: string]: string; } | undefined) => Promise<unknown>)'.
    Type '{ setup: () => Promise<void>; teardown: () => Promise<void>; }' is not assignable to type '(state: string, params?: { [name: string]: string; } | undefined) => Promise<unknown>'.
      Type '{ setup: () => Promise<void>; teardown: () => Promise<void>; }' provides no match for the signature '(state: string, params?: { [name: string]: string; } | undefined): Promise<unknown>'.
```

### Steps to reproduce

```ts
import { Verifier } from '@pact-foundation/pact';

const verifier = new Verifier({
  stateHandlers: {
    'name of the stateHandler': { setup: async () => {}, teardown: async () => {} } // this line produces a TypeScript error
  }
});
```

--- a/src/dsl/verifier/proxy/stateHandler/setupStates.ts
+++ b/src/dsl/verifier/proxy/stateHandler/setupStates.ts
@@ -1,4 +1,3 @@
-import type { JsonMap } from '../../../../common/jsonTypes';
 import logger from '../../../../common/logger';
 import type {
   ProviderState,
@@ -22,7 +21,7 @@ const transformStateFunc = (fn: StateHandler): StateFuncWithSetup =>
 export const setupStates = (
   state: ProviderState,
   config: ProxyOptions,
-): Promise<JsonMap | undefined> => {
+): ReturnType<StateFunc> => {
   logger.debug(`setting up state '${JSON.stringify(state)}'`);
 
   const handler = config.stateHandlers

--- a/src/dsl/verifier/proxy/types.ts
+++ b/src/dsl/verifier/proxy/types.ts
@@ -30,7 +30,8 @@ export type StateAction = 'setup' | 'teardown';
  * Respond to the state setup event, optionally returning a map of provider
  * values to dynamically inject into the incoming request to test
  */
-export type StateFunc = (parameters?: AnyJson) => Promise<JsonMap | undefined>;
+// biome-ignore lint/suspicious/noConfusingVoidType: state handlers may intentionally resolve without a value.
+export type StateFunc = (parameters?: AnyJson) => Promise<JsonMap | void>;
 
 /**
  * Respond to the state setup event, with the ability to hook into the setup/teardown

--- a/src/dsl/verifier/types.ts
+++ b/src/dsl/verifier/types.ts
@@ -1,5 +1,4 @@
 import type { VerifierOptions as PactCoreVerifierOptions } from '@pact-foundation/pact-core';
-import type { MessageProviderOptions } from '../options';
 
 import type { ProxyOptions } from './proxy/types';
 
@@ -14,5 +13,4 @@ export type PactNodeVerificationExcludedOptions = Pick<
 >;
 
 export type VerifierOptions = PactNodeVerificationExcludedOptions &
-  ProxyOptions &
-  Partial<MessageProviderOptions>;
+  ProxyOptions;

--- a/src/dsl/verifier/verifier.spec.ts
+++ b/src/dsl/verifier/verifier.spec.ts
@@ -55,6 +55,22 @@ describe('Verifier', () => {
         expect(v).toHaveProperty('config.stateHandlers');
         expect(v).toHaveProperty('config.requestFilter');
       });
+
+      it('accepts state handlers with separate setup and teardown functions', () => {
+        const stateHandler = {
+          setup: vi.fn(async () => {}),
+          teardown: vi.fn(async () => {}),
+        };
+
+        v = new Verifier({
+          providerBaseUrl,
+          stateHandlers: {
+            [state]: stateHandler,
+          },
+        });
+
+        expect(v).toHaveProperty(`config.stateHandlers.${state}`, stateHandler);
+      });
     });
   });
 

